src/TableauExtract.cc

```diff
@@ -104,7 +105,11 @@ void Extract::HasTable(const FunctionCallbackInfo<Value>& args) {
 }
 
 void Extract::AddTable(const FunctionCallbackInfo<Value>& args) {
-  NodeTde::Table::NewInstanceFromDefinition(args);
+  try {
+    NodeTde::Table::NewInstanceFromDefinition(args);
+  } catch (const Tableau::TableauException& e) {
+    THROW_TABLEAU_EXCEPTION(e);
```

Comment:
argh,  this pr was not ready, sorry

I just wanted to push the readme file

will cancel this
On Jun 17, 2016 9:29 PM, "Eric Peterson" <notifications@github.com> wrote:

> In src/TableauExtract.cc
> <https://github.com/tableau-mkt/node-tableau-sdk/pull/6#discussion_r67563206>
> :
>
> > @@ -104,7 +105,11 @@ void Extract::HasTable(const FunctionCallbackInfo<Value>& args) {
> >  }
> >
> >  void Extract::AddTable(const FunctionCallbackInfo<Value>& args) {
> > -  NodeTde::Table::NewInstanceFromDefinition(args);
> > +  try {
> > +    NodeTde::Table::NewInstanceFromDefinition(args);
> > +  } catch (const Tableau::TableauException& e) {
> > +    THROW_TABLEAU_EXCEPTION(e);
>
> A quick test for this would be nice. Something like...
>
> var expect = chai.expect;// ...it('throws error when adding invalid table name', function () {
>   // Create an extract.
>   expectedPath = targetDir + '/mocha-add-invalid-table.tde';
>   extract = tableau.dataExtract(expectedPath);
>
>   // Create a table definition.
>   tableDef = tableau.tableDefinition();
>   tableDef.addColumn('Column', tableau.enums.type('Boolean'));
>
>   // Attempt to add a table with a name other than Extract.
>   expect(extract.addTable.bind(extract, 'Invalid', tableDef))
>     .to.throw();
> });
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tableau-mkt/node-tableau-sdk/pull/6/files/1cda6ca8a5e4551b6503ac96eb3e307d0533d860#r67563206>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAFB-q-bgpoMQyavMYU8p9xIhCB1tO2Bks5qMvWrgaJpZM4I4KMu>
> .
>